runtime/interpreter/interpreter.go

```diff
@@ -2573,11 +2573,188 @@ func init() {
 			panic(fmt.Sprintf("missing converter for number type: %s", numberType))
 		}
 	}
+
+	// We assign this here because it depends on the interpreter, so this breaks the initialization cycle
+	defineBaseValue(
+		baseActivation,
+		"DictionaryType",
+		NewHostFunctionValue(
+			func(invocation Invocation) Value {
+
+				keyType := invocation.Arguments[0].(TypeValue).Type
+
+				// if the given key is not a valid dictionary key, it wouldn't make sense to create this type
+				if keyType == nil ||
+					!sema.IsValidDictionaryKeyType(invocation.Interpreter.ConvertStaticToSemaType(keyType)) {
+					return NilValue{}
+				}
+
+				return NewSomeValueOwningNonCopying(TypeValue{
+					Type: DictionaryStaticType{
+						KeyType:   keyType,
+						ValueType: invocation.Arguments[1].(TypeValue).Type,
+					}})
```

Comment:
Maybe declare all inputs at the top, so it is easier visible:

```suggestion
				keyType := invocation.Arguments[0].(TypeValue).Type
				valueType := invocation.Arguments[1].(TypeValue).Type

				// if the given key is not a valid dictionary key, it wouldn't make sense to create this type
				if keyType == nil ||
					!sema.IsValidDictionaryKeyType(invocation.Interpreter.ConvertStaticToSemaType(keyType)) {
					return NilValue{}
				}

				return NewSomeValueOwningNonCopying(
					TypeValue{
						Type: DictionaryStaticType{
							KeyType:   keyType,
							ValueType: valueType,
						},
					},
				)
```